File: 2/2_cuatrimestre/PDOO/practicas/practica 4/deepSpace_4/src/deepspace/BetaPowerEfficientSpaceStation.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author carlos
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation {
    private final float EXTRAEFFICIENCY;
    private Dice dice;
    public BetaPowerEfficientSpaceStation(SpaceStation station){
        super(station);
        EXTRAEFFICIENCY = 1.10F;
        dice = new Dice();
        
    }
    
    public float fire(){
        if(dice.extraEfficiency())
            return super.fire()*EXTRAEFFICIENCY;
        else
            return super.fire();
    }
    
    public BetaPowerEfficientSpaceStationToUI getUIversion(){
            return new BetaPowerEfficientSpaceStationToUI(this);
    }
    
    @Override
    public String toString(){
        return getUIversion().toString();
    }
}

File: 2/2_cuatrimestre/PDOO/practicas/practica 4/deepSpaceRuby/BetaPowerEfficientSpaceStation.rb
require_relative 'Dice'
require_relative 'BetaPowerEfficientSpaceStationToUI'
require_relative 'PowerEfficientSpaceStation'

module Deepspace
    class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
        def initialize (station)
            @EXTRAEFFICIENCY = 1.2
            @dice = Dice.new
            super
        end


        def fire
            if (@dice.extraEfficiency)
                super * @EXTRAEFFICIENCY
            else
                super
            end
        end

        
        def getUIversion
            BetaPowerEfficientSpaceStationToUI.new(self)
        end

        def to_s
            return getUIversion.to_s
        end
    end
end